lxd/device/disk.go

```diff
@@ -157,6 +157,40 @@ func (d *disk) sourceIsLocalPath(source string) bool {
 	return true
 }
 
+// Check that unshared custom storage block volumes are not added to profiles or multiple instances.
+func (d *disk) checkBlockVolSharing(instanceType instancetype.Type, projectName string, volume *api.StorageVolume) error {
+	contentType, err := storagePools.VolumeContentTypeNameToContentType(volume.ContentType)
+	if err != nil {
+		return err
+	}
+
+	// Skip the checks if the volume is set to be shared or is not a block volume.
+	if contentType != cluster.StoragePoolVolumeContentTypeBlock || shared.IsTrue(volume.Config["security.shared"]) {
```

Comment:
Can you just use cluster.StoragePoolVolumeContentTypeNameBlock here and avoid the conversion above?